src/main/java/nextstep/courses/domain/Status.java

```diff
@@ -0,0 +1,34 @@
+package nextstep.courses.domain;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public enum Status {
+    READY("READY"), RECRUIT("RECRUIT"), END("END");
+
+    private final String name;
+
+    private static final Map<String, Status> STATUS_MAP =
+            Collections.unmodifiableMap(Stream.of(values())
+                    .collect(Collectors.toMap(Status::toString, Function.identity())));
+
+    Status(String name) {
+        this.name = name;
+    }
+
+    public static boolean isNotRecruit(Status status) {
+        return RECRUIT != status;
+    }
+
+    public static Status find(String status) {
+        if (STATUS_MAP.containsKey(status)) {
+            return STATUS_MAP.get(status);
+        }
+        return null;
```

Comment:
`null`을 반환하신다면 굳이 `containsKey`를 확인할 필요는 없을 것 같습니다 😄 
아마 `toMap`으로 변환할 때 `HashMap`으로 변환할텐데요 🤔 
값이 없으면 `null`로 반환하고 있습니다 
참고만 해주시면 될 것 같습니다 😄 

  ```java
  /**
     * Returns the value to which the specified key is mapped,
     * or {@code null} if this map contains no mapping for the key.
     *
     * <p>More formally, if this map contains a mapping from a key
     * {@code k} to a value {@code v} such that {@code (key==null ? k==null :
     * key.equals(k))}, then this method returns {@code v}; otherwise
     * it returns {@code null}.  (There can be at most one such mapping.)
     *
     * <p>A return value of {@code null} does not <i>necessarily</i>
     * indicate that the map contains no mapping for the key; it's also
     * possible that the map explicitly maps the key to {@code null}.
     * The {@link #containsKey containsKey} operation may be used to
     * distinguish these two cases.
     *
     * @see #put(Object, Object)
     */
    public V get(Object key) {
        Node<K,V> e;
        return (e = getNode(hash(key), key)) == null ? null : e.value;
    }
```